[AOTI] Split aoti_runtime/model.h to prepare for model static linking

yushangdi opened the pull request:
Summary:
Prepare for https://github.com/pytorch/pytorch/pull/157129.

We split the file so we can re-use `model.h` part for codegen a separate header for each model in static linkage.

Test Plan:
CI

Rollback Plan:

Differential Revision: D77761249

desertfire commented on torch/csrc/inductor/aoti_runtime/model.h:
```diff
@@ -1,732 +1,9 @@
-
-} // anonymous namespace
+#include <torch/csrc/inductor/aoti_runtime/model_base.h>
 
```
Please also copy the same WARNING here. All the header files in aoti_runtime have this warning just to play safe.

yushangdi committed:
[AOTI] Split aoti_runtime/model.h to prepare for model static linking (#157592)

Summary:

Prepare for https://github.com/pytorch/pytorch/pull/157129.

We split the file so we can re-use `model.h` part for codegen a separate header for each model in static linkage.

Test Plan:
CI

Rollback Plan:

Reviewed By: desertfire

Differential Revision: D77761249
